pkg/asset/installconfig/aws/validation.go

```diff
@@ -0,0 +1,153 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/ipnet"
+	"github.com/openshift/installer/pkg/types"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+)
+
+// Validate executes platform-specific validation.
+func Validate(ctx context.Context, meta *Metadata, config *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	if config.Platform.AWS != nil {
```

Comment:
a96182f3f4 puts a guard in `InstallConfig.platformValidation` to avoid calling this `Validate` unless we're on AWS.  And `platformValidation` is called after `ValidateInstallConfig` in `InstallConfig.finish`.  So I think we want:

```go
if config.Platform.AWS == nil {
  return field.Required(field.NewPath("platform", "aws"), "AWS validation requires an AWS platform configuration")
}
```
or some such.

Reply:
hmm, i don't know who i feel about that, I can still do it.

but would like to keep the ties like this function is being called after static validation to minimum.

Reply:
@wking fixed in https://github.com/openshift/installer/compare/a96182f3f4d7c7b69bd674bd918dd72a532f6ce4..9ce0bb39f5cbc2d581705794c8e16933cae350d4